lib/asn1integer.dart

```diff
@@ -3,86 +3,90 @@ part of asn1lib;
  * ASN1Integer encoding / decoding.
  *
  * Note that asn1 integers can be arbitrary precision.
- * TODO: convert use of BigInteger to dart sdk BigInt class.
  */
 
 class ASN1Integer extends ASN1Object {
-  var intValue;
+  BigInt _intValue;
 
-  ASN1Integer(this.intValue, {tag: INTEGER_TYPE}) : super(tag: tag);
+  ASN1Integer(this._intValue, {tag: INTEGER_TYPE}) : super(tag: tag);
 
   ASN1Integer.fromBytes(Uint8List bytes) : super.fromBytes(bytes) {
-    intValue = decodeInteger(this.valueBytes());
+    _intValue = bytes2BigInt(this.valueBytes());
   }
 
-  BigInteger get valueAsBigInteger {
-    if (intValue is BigInteger) return intValue;
-    if (isEncoded) return new BigInteger(valueBytes());
-    return new BigInteger(intValue);
-  }
+  int get intValue => valueAsBigInteger.toInt();
 
-  BigInteger get valueAsPositiveBigInteger {
-    if (!isEncoded) _encode();
-    return new BigInteger.fromBytes(1, valueBytes());
-  }
+  BigInt get valueAsBigInteger => _intValue;
 
   @override
   Uint8List _encode() {
-    var t = encodeIntValue(this.intValue);
+    var t = encodeBigIntValue(this._intValue);
```

Comment:
For string mode, shouldn't this be `List<int>` instead of `var`?

Reply:
strong mode? ok - makes sense